core/src/main/scala/akka/kafka/internal/DefaultProducerStage.scala

```diff
@@ -29,28 +29,31 @@ import scala.util.{Failure, Success, Try}
 private[kafka] class DefaultProducerStage[K, V, P, IN <: Envelope[K, V, P], OUT <: Results[K, V, P]](
     val closeTimeout: FiniteDuration,
     val closeProducerOnStop: Boolean,
-    val producerProvider: () => Producer[K, V]
+    val producerProvider: Unit => Producer[K, V]
 ) extends GraphStage[FlowShape[IN, Future[OUT]]]
-    with ProducerStage[K, V, P, IN, OUT] {
+    with ProducerStage[K, V, P, IN, OUT, Unit] {
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
-    new DefaultProducerStageLogic(this, producerProvider(), inheritedAttributes)
+    new DefaultProducerStageLogic(this, producerProvider, inheritedAttributes) {
+      override protected var producer: Producer[K, V] = producerProvider(())
+    }
 }
 
 /**
  * Internal API.
  *
  * Used by [[DefaultProducerStage]], extended by [[TransactionalProducerStageLogic]].
  */
-private class DefaultProducerStageLogic[K, V, P, IN <: Envelope[K, V, P], OUT <: Results[K, V, P]](
-    stage: ProducerStage[K, V, P, IN, OUT],
-    producer: Producer[K, V],
+private abstract class DefaultProducerStageLogic[K, V, P, IN <: Envelope[K, V, P], OUT <: Results[K, V, P], S](
```

Comment:
The new type parameter is not obvious, add a `@tparam` to the Scaladoc.

Reply:
This type param is no longer applicable.